### PR TITLE
Ractor usability improvements

### PR DESCRIPTION
**Pull request by zverok**
Small improvements to `Ractor` usability:

* alias `#to_s` to `#inspect`:
  ```
  ./ruby --disable-gems -e 'r = Ractor.new{}; puts "Created a ractor: #{r}"'
  # before:
  Created a ractor: #<Ractor:0x00005600e210c2d0>
  # after:
  Created a ractor: #<Ractor:#2 -e:1 blocking>
  ```
* expose `Ractor#status`:
  ```
  $ ./ruby --disable-gems -e "p Ractor.new{}.status"
  :blocking
  ```

TODO: 
1. ~I haven't found in the repo tests for basic ractor behaior, am I missing something? So I tested it only manually (though I doubt anything might be broken)~
2. ~Suddenly, I can't find a better way for producing symbols from `__builtin_cexpr!`~
@ko1 @marcandre @eregon

### CODE REVIEW

**Comment by nobu** [comment]
`rb_sym_intern_ascii_cstr()` needs "internal/symbol.h".

```diff
diff --git i/common.mk w/common.mk
index 607ad861aa3..d99703a60bc 100644
--- i/common.mk
+++ w/common.mk
@@ -10214,4 +10214,5 @@ ractor.$(OBJEXT): $(top_srcdir)/internal/static_assert.h
 ractor.$(OBJEXT): $(top_srcdir)/internal/string.h
 ractor.$(OBJEXT): $(top_srcdir)/internal/struct.h
+ractor.$(OBJEXT): $(top_srcdir)/internal/symbol.h
 ractor.$(OBJEXT): $(top_srcdir)/internal/vm.h
 ractor.$(OBJEXT): $(top_srcdir)/internal/warnings.h
diff --git i/ractor.c w/ractor.c
index 08150f49a3e..841920fd352 100644
--- i/ractor.c
+++ w/ractor.c
@@ -13,4 +13,5 @@
 #include "internal/rational.h"
 #include "internal/struct.h"
+#include "internal/symbol.h"
 #include "variable.h"
 #include "gc.h"
diff --git i/ractor.rb w/ractor.rb
index dc097171b08..e2172671590 100644
--- i/ractor.rb
+++ w/ractor.rb
@@ -201,7 +201,6 @@
   #
   def status
-    # FIXME: why rb_sym_intern_ascii_cstr(...) does not compile?
     __builtin_cexpr! %q{
-      rb_str_intern(rb_str_new2(ractor_status_str(RACTOR_PTR(self)->status_)))
+      rb_sym_intern_ascii_cstr(ractor_status_str(RACTOR_PTR(self)->status_))
     }
   end
```

**Comment by zverok** [comment]
@nobu Thanks, I fixed and force-pushed the code.

**Comment by marcandre** [comment]
>     1. I haven't found in the repo tests for basic ractor behaior, am I missing something? So I tested it only manually (though I doubt anything might be broken)

It's in `bootstraptest/test_ractor.rb` (run it with `make bstrap-ruby`) or you could start `spec/rubyspec/core/ractor/`...

File: ractor.rb [hunk 1]
```diff
@@ -185,10 +185,26 @@ def inspect
...
   def name
     __builtin_cexpr! %q{ RACTOR_PTR(self)->name }
   end
 
+  # Returns the current status of the ractor.
+  #
+  # Statuses are:
+  # * <tt>:created</tt> -- the ractor is just created
+  # * <tt>:running</tt> -- the ractor is running
+  # * <tt>:blocking</tt> -- the ractor is waiting on Racor.receive or Ractor.yield
```
Comment (marcandre) [inline ractor.rb:199]: ^Racor^Ractor

File: ractor.rb [hunk 2]
```diff
@@ -185,10 +185,26 @@ def inspect
...
 
+  alias to_s inspect
+
   def name
     __builtin_cexpr! %q{ RACTOR_PTR(self)->name }
   end
 
+  # Returns the current status of the ractor.
+  #
+  # Statuses are:
```
Comment (marcandre) [inline ractor.rb:196]: I'd reword as "Possible values are:"

**Review by marcandre — COMMENTED** [review]
Good changes!
Some nitpicks on wording

**Comment by ko1** [comment]
`Ractor#status` is not provided intentionally because someone can write such code:

```ruby
r = Ractor.new{...}
sleep 0.1 while r.status == :running
```

**Comment by zverok** [comment]
@ko1 This argument seems not very strong for me:

1. First, as status included in `#inspect`, it doesn't seem very "polite" for the user: "there IS a status, we know it, but we don't tell you"; I don't feel like concealing of the information which is obviously there in order to educate user is in the spirit of Ruby
2. Second, as status included into `#inspect`, your "bad" code can be easily rewritten as
   ```ruby
   sleep 0.1 while r.inspect.include?('running')
   ```
3. Finally, availability of the `#status` is a great help in debuging, investigating and educating

**Comment by marcandre** [comment]
Agree with @zverok 

You are being too nice @ko1 😅
Let users shoot themselves in the foot.

**Comment by zverok** [comment]
@marcandre Thanks, I've updated the PR

**Comment by nobu** [comment]
You are missing the "common.mk" part.

**Comment by zverok** [comment]
@nobu Sorry, added (locally it somehow worked without this change), and rebased from the latest master.

**Comment by marcandre** [comment]
Very odd, there's a failure on MacOS...

**Comment by zverok** [comment]
@nobu Do you know why macOS builds fail?.. Other than that, the PR is ready for merge.

**Comment by marcandre** [comment]
> Other than that, the PR is ready for merge.

Still need to convince @ko1 or Matz...

**Comment by ko1** [comment]
> @ko1 This argument seems not very strong for me:
> 
> 1. First, as status included in `#inspect`, it doesn't seem very "polite" for the user: "there IS a status, we know it, but we don't tell you"; I don't feel like concealing of the information which is obviously there in order to educate user is in the spirit of Ruby

So you can find the status in `#inspect`.

> 2. Second, as status included into `#inspect`, your "bad" code can be easily rewritten as
>    ```ruby
>    sleep 0.1 while r.inspect.include?('running')
>    ```

Yes.  You can shoot your own foot.

> 3. Finally, availability of the `#status` is a great help in debuging, investigating and educating

again, you can find the status in `#inspect`.

**Comment by zverok** [comment]
Whatever

**Comment by marcandre** [comment]
I committed the alias for `to_s` in 8f2031a06725
